app/Http/Controllers/Api/CtsController.php

```diff
@@ -0,0 +1,80 @@
+<?php
+
+namespace App\Http\Controllers\Api;
+
+use App\Repositories\Cts\BookingRepository;
+use Carbon\Carbon;
+use Illuminate\Http\Request;
+use Illuminate\Http\Response;
+use Illuminate\Support\Facades\RateLimiter;
+
+class CtsController
+{
+    public function __construct(private BookingRepository $bookingRepository) {}
+
+    public function getBookings(Request $request)
+    {
+        if (app()->environment() !== 'development' && RateLimiter::tooManyAttempts('get-bookings:'.$request->ip(), 1)) {
+            $seconds = RateLimiter::availableIn('get-bookings:'.$request->ip());
+
+            return response()->json([
+                'message' => 'Too many requests. Please try again after '.$seconds.' seconds.',
+            ], Response::HTTP_TOO_MANY_REQUESTS)
+                ->header('Retry-After', $seconds)
+                ->header('X-RateLimit-Reset', now()->addSeconds($seconds)->getTimestamp());
+        }
+
+        RateLimiter::hit('get-bookings:'.$request->ip(), 300); // 5 minutes
+
+        $date = $request->get('date', null);
+
+        // Validate date, default to today
+        if ($date) {
+            try {
+                $date = Carbon::parse($date);
+
+                if ($date->isPast() && $date->diffInDays(Carbon::now()) > 30) {
+                    return response()->json([
+                        'message' => 'Date is too far in the past. Please use a date within the last 30 days. Oldest date allowed: '.Carbon::now()->subDays(30)->toDateString(),
+                    ], Response::HTTP_BAD_REQUEST);
+                }
+
+            } catch (\Carbon\Exceptions\InvalidFormatException $e) {
+                return response()->json([
+                    'message' => 'Invalid date format. Please use YYYY-MM-DD.',
+                ], Response::HTTP_BAD_REQUEST);
+            }
+
+        } else {
+            $date = Carbon::now()->startOfDay();
+        }
```

Comment:
You could simplify this code by defining `$date` above on line 31, and then taking a user requested date accordingly.